bndtools.jareditor/src/bndtools/jareditor/internal/TemporaryProjectFilter.java

```diff
@@ -0,0 +1,27 @@
+package bndtools.jareditor.internal;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.jface.viewers.ViewerFilter;
+
+public class TemporaryProjectFilter extends ViewerFilter {
+
+	@Override
+	public boolean select(Viewer viewer, Object parentElement, Object element) {
+		if (element instanceof IJavaProject) {
+			element = ((IJavaProject) element).getProject();
+		}
+
+		return !Optional.ofNullable(element)
+			.filter(IProject.class::isInstance)
+			.map(IProject.class::cast)
+			.filter(Objects::nonNull)
```

Comment:
Why do you need the non-null filter? The `Class.isInstance` will already filter out null values.

Reply:
Good point, I can remove it.